src/plugins/home/public/application/components/apm_fleet/index.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import {
+  EuiButton,
+  EuiCard,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiImage,
+  EuiLoadingSpinner,
+  EuiPanel,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { getServices } from '../../kibana_services';
+
+const CentralizedContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+interface APIResponse {
+  hasData: boolean;
+}
+
+export function APMFleet() {
+  const { http, getBasePath, uiSettings } = getServices();
+  const isDarkTheme = uiSettings.get('theme:darkMode');
+  const basePath = getBasePath();
+  const [data, setData] = useState<APIResponse | undefined>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    async function fetchData() {
+      setIsLoading(true);
+      try {
+        const response = await http.get('/api/apm/fleet/has_data');
+        setData(response as APIResponse);
```

Comment:
Having a component in the `home` plugin performing an http call to an endpoint exposed by another plugin, especially without having an explicit dependency against the second plugin, seems like a big code smell.

Reply:
Please see: https://github.com/elastic/kibana/pull/100816#discussion_r644203888

Reply:
@pgayvallet Completely agree with your points but I'm not sure there is a better way, except rewriting the tutorials or moving away from them. The tutorial DSL is overly limiting and not easy to work around since it's rendered server side.

As an aside, I think this can be a good learning experience within Kibana. We should only build restrive frameworks like this, when there are active maintainers that are given time to refactor as new use cases emerge. 
The other option is to have very flexible frameworks where anything goes basically (convention over configuration).

For more than 2 years the APM team have been unable to make meaningful changes to the tutorial (and yes, we have decided not to spend time refactoring the tutorial codebase even though that might have been time well spent).

Reply:
> This is currently abandonware

I (unfortunately) won't argue on this regarding the `home` plugin.

One possible workaround would be to add a `registerCustomStatusCheck(name, fn)` for https://github.com/elastic/kibana/pull/100816#discussion_r643965851 and `registerCustomComponent(name, component)` for this call. That would allow to have the component and status check be registered by `apm` while keeping the `apm` -> `home` dependency in the right order. The server-side tutorial would just register the potential name of the status and/or component (as already done in current PR)

However I agree that
1. this is some non-negligible refactoring for a currently isolated use-case
2. this splits the tutorial registration in two part, as the tutorial registered on the server-side requires some additional calls from the client-side plugin to be properly displayed, so even performing this refactoring would lead to a flawed solution

So, I'm mixed. @joshdover, wdyt? should we just accept that it's not worth the cost and merged as is?

Reply:
I'm mixed as well, and I think the answer likely depends on what the long-term plans with the tutorial directory are. @thesmallestduck is this something you could help us track down in the updated onboarding UX flows? Do we plan to keep these tutorials in the long-term or will those exclusively live in the Integrations UI?

Reply:
Thanks for the answer @pgayvallet. I'll let @cauemarcondes look into your proposed solution. We also want to do this the right way even if it's more work. Still, cost-benefit and depends on the long term plans for tutorials.

Reply:
@pgayvallet @joshdover @sqren I followed what was suggested and created two registration functions (`registerCustomStatusCheck` and `registerCustomComponent`). Both the callback function and the custom component were moved to inside `apm` so we won't need to add specific apm checks/component inside the home tutorial.